agent.go

```diff
@@ -583,7 +583,7 @@ func (ep *endpoint) deleteDriverInfoFromCluster() error {
 }
 
 func (ep *endpoint) addServiceInfoToCluster(sb *sandbox) error {
-	if ep.isAnonymous() && len(ep.myAliases) == 0 || ep.Iface().Address() == nil {
+	if ep.isAnonymous() && len(ep.myAliases) == 0 || ep.Iface() == nil || ep.Iface().Address() == nil {
```

Comment:
Could use some input here; I started with this, but noticed there's quite some places where a "fluent" interface is used (`foo.Iface().SomeFunction()`), so having to check in _all_ places is likely problematic.